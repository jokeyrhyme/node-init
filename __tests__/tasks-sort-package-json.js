/* @flow */
'use strict';

jest.mock('execa');
jest.mock('update-json-file');

test('fn()', async () => {
  const { fn } = require('../lib/tasks/sort-package-json.js');
  await fn({ cwd: process.cwd(), scope: '' });

  expect(require('execa').mock.calls).toMatchSnapshot();

  // $FlowFixMe: ignore .mock's missing type
  expect(require('update-json-file').mock.calls.length).toBe(1);

  // $FlowFixMe: ignore .mock's missing type
  const editor = require('update-json-file').mock.calls[0][1];
  expect(editor({})).toMatchSnapshot();
});
