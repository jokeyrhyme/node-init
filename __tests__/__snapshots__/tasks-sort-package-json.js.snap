// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn() 1`] = `
Array [
  Array [
    "npm",
    Array [
      "uninstall",
      "--save-dev",
      "fixpack",
    ],
    Object {
      "cwd": "/home/ron/GitHub/jokeyrhyme/node-init.js",
    },
  ],
]
`;

exports[`fn() 2`] = `
Object {
  "scripts": Object {
    "pretest": "npm run sort-package-json",
    "sort-package-json": "npx -q sort-package-json",
  },
}
`;
